sdk/core/Azure.Core.Experimental/api/Azure.Core.Experimental.netstandard2.0.cs

```diff
@@ -120,3 +120,73 @@ public ProvisionableTemplateAttribute(string resourceName) { }
+    {
+        string[] Scopes { get; }
+        object CloneWithAdditionalScopes(string[] additionalScopes);
+    }
+    public partial interface ITokenContext
+    {
+    }
+    public partial interface ITokenProvider
+    {
+        object CreateContext(System.Collections.Generic.IReadOnlyDictionary<string, object> properties);
+        System.ClientModel.Token GetAccessToken(object context, System.Threading.CancellationToken cancellationToken);
+        System.Threading.Tasks.ValueTask<System.ClientModel.Token> GetAccessTokenAsync(object context, System.Threading.CancellationToken cancellationToken);
+    }
+    public abstract partial class RefreshableToken : System.ClientModel.Token
+    {
+        public RefreshableToken(string tokenValue, string tokenType, System.DateTimeOffset expiresOn, System.DateTimeOffset? refreshOn = default(System.DateTimeOffset?)) : base (default(string), default(string), default(System.DateTimeOffset), default(System.DateTimeOffset?)) { }
+        public abstract System.Threading.Tasks.Task RefreshAsync(System.Threading.CancellationToken cancellationToken);
+    }
+    public partial class Token
+    {
+        public Token(string tokenValue, string tokenType, System.DateTimeOffset expiresOn, System.DateTimeOffset? refreshOn = default(System.DateTimeOffset?)) { }
+        public System.DateTimeOffset ExpiresOn { get { throw null; } protected set { } }
+        public System.DateTimeOffset? RefreshOn { get { throw null; } protected set { } }
+        public string TokenType { get { throw null; } protected set { } }
+        public string TokenValue { get { throw null; } protected set { } }
+    }
+    public abstract partial class TokenProvider<TContext> : System.ClientModel.ITokenProvider where TContext : System.ClientModel.ITokenContext
+    {
+        protected TokenProvider() { }
+        public abstract TContext CreateContext(System.Collections.Generic.IReadOnlyDictionary<string, object> properties);
+        public abstract System.ClientModel.Token GetAccessToken(TContext context, System.Threading.CancellationToken cancellationToken);
+        public abstract System.Threading.Tasks.ValueTask<System.ClientModel.Token> GetAccessTokenAsync(TContext context, System.Threading.CancellationToken cancellationToken);
+        object System.ClientModel.ITokenProvider.CreateContext(System.Collections.Generic.IReadOnlyDictionary<string, object> properties) { throw null; }
+        System.ClientModel.Token System.ClientModel.ITokenProvider.GetAccessToken(object context, System.Threading.CancellationToken cancellationToken) { throw null; }
+        System.Threading.Tasks.ValueTask<System.ClientModel.Token> System.ClientModel.ITokenProvider.GetAccessTokenAsync(object context, System.Threading.CancellationToken cancellationToken) { throw null; }
+    }
+}
+namespace System.ClientModel.Auth
+{
+    public partial class OAuth2BearerTokenAuthenticationPolicy : System.ClientModel.Primitives.PipelinePolicy
```

Comment:
We should pick whether we're using `Auth` or `Authentication` and be consistent.

Reply:
I think we might want to move everything but `Token` and `RefreshableToken` into this namespace too.  I don't think most people would ever have to touch those interfaces unless you were implementing an auth provider.

If we don't have enough types to justify a namespace, we could also move them all into `System.ClientModel.Primitives` too.  We used to have a hierarchy like:

- `Azure`: For end users
- `Azure.Core`: For library authors
- `Azure.Core.*`: For our team

But we simplified that a bit with SCM into just:

- `System.ClientModel`: For end users
- `System.ClientModel.Primitives`: For library authors and our team

Reply:
I chose Auth for now, since we have a handful of types and may have more over time.